File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        mpp = {0:-1}
        sum = 0
        for i in range(0, len(nums)):
            sum+= nums[i]
            sum%=k
            
            if sum not in mpp:
                mpp[sum]=i
            elif i -mpp[sum]>1:
                return True
            
        return False

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) 
    {
        int n = nums.size();
        
        unordered_map<int,int> mpp;
        
        mpp.insert({0,-1});
        
        int sum = 0;
        
        for(int i=0;i<n;i++)
        {
            sum+= nums[i];
            sum%=k;
            
            if(mpp.find(sum)==mpp.end())
            {
               mpp[sum]=i;
            }
            else if(i-mpp[sum]>1)
                return true;
                
        }
        
        return false;
    }
};